### PR TITLE
Changed FHIR server for integration tests to pass

### PR DESCRIPTION
**Pull request by akanduru**
These changes are on top of PR #125 and PR #113.

I changed the FHIR server in the integration tests to http://hapi.fhir.org. I hope HAPI folks don't curse me! 

Ideally we should mock the FHIR server calls to shield the tests from broken server. I thought a quick fix is needed to address the pending PRs.

### CODE REVIEW

File: src/adapters/node.js [hunk 1]
```diff
@@ -8,11 +7,20 @@
         defer: Q.defer,
         http: function(args) {
             var deff = Q.defer();
-            args.body = args.data;
+            if(args.data && typeof args.data === "string") {
+              // Setting args.json requires args.body to be a JSON as per request docs.
+              args.body = JSON.parse(args.data);
```
Comment (mike1pol) [inline src/adapters/node.js]: Need to use `try {} catch (e) {}` construction for JSON.parse, string may is not JSON format

Comment (mike1pol) [reply src/adapters/node.js]: You can make custom error if string is wrong JSON

Comment (akanduru) [reply src/adapters/node.js]: If we catch the error we should through it back to caller. I am trying to understand the suggestion. Are you saying we should give the caller a better message and/or log the error?

Comment (mike1pol) [reply src/adapters/node.js]: Yes, we can send better message for caller.
Like this:
```
try {
  args.body = JSON.parse(args.data);
} catch (e) {
  throw new Error('...string is not JSON', e);
}
```

Comment (akanduru) [reply src/adapters/node.js]: That will mask the original stacktrace...

Comment (mike1pol) [reply src/adapters/node.js]: why?

if we don't catch error:
```
SyntaxError: Unexpected token a in JSON at position 0
    at JSON.parse (<anonymous>)
    at http (/Users/mike1pol/work/hs/fhir.js/src/adapters/node.js:12:20)
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:9:67
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/search.js:182:20
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:25:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:68:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
```
if we catch error
```
Error: data is not a JSON
    at http (/Users/mike1pol/work/hs/fhir.js/src/adapters/node.js:15:11)
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:9:67
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/search.js:182:20
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:25:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:68:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:27:20

```

Comment (akanduru) [reply src/adapters/node.js]: The line 12 is the original error. If there are more lines added to try block, then it is a guess work to pin point the error.

In any case it is not my main concern. I can go with the change. Please let me know, if you have other suggestions.

**Pull request merged**